src/Framework/EventFile.cxx

```diff
@@ -194,7 +198,19 @@ bool EventFile::nextEvent(bool storeCurrentEvent) {
     // if we are reading, move the pointer
     if (!isOutputFile_) {
       if (ientry_ + 1 >= entries_) {
-        return false;
+		if ( isOverlayFile_ ) {
+		  // reset the event counter: reuse events from start of pileup tree
+
+		  /* Here I would have liked to inform the user that a reset happens
+			 but it seems like only Exceptions are foreseen from this class? 
+			 ldmx_log(info)
+		  std::cout  << "Reached end of pileup tree at entry " <<
+			ientry_ << "; resetting event counter!" << std::endl;
+		  */
```

Comment:
The reason the message passed to `ldmx_log(info)` here is "lossed" is because it is a macro that looks for a variable called `theLog_` to write the message to. You could enable logging for the EventFile class using the enableLogging macro, but I also don't think this is necessary.

Reply:
Alright. Deleted the comment